src/sentry/integrations/notify_disable.py

```diff
@@ -37,15 +38,21 @@ def get_subject(integration_name: str) -> str:
 
 def notify_disable(
     organization: Organization,
-    integration_name: str,
+    integration: Union[SentryApp, RpcIntegration],
     redis_key: str,
     project: Union[str, None] = None,
 ):
 
     integration_link = get_url(
         organization,
         get_provider_type(redis_key),
-        integration_name,
+        integration.slug if hasattr(integration, "slug") else integration.provider,
+    )
+
+    integration_name = (
```

Comment:
@chloeho7 can we expose methods on the `SentryApp` and `RpcIntegration` to return the slug/name so this function doesn't have to have this kind of logic?

Reply:
Sure, now using `.get_provider()` from `RpcIntegration`  instead of `hasattr()`

Reply:
Damn typing really doesn't like this, it'll have to be rewritten a bit.